lisa/wlgen/workload.py

```diff
@@ -133,16 +133,19 @@ def run(self, cpus=None, cgroup=None, background=False, as_root=False, timeout=N
 
         logger.info("Execution start: %s", _command)
 
+        # Just return the Popen handle
         if background:
-            target.background(_command, as_root=as_root)
-        else:
-            self.output = target.execute(_command, as_root=as_root, timeout=timeout)
-            logger.info("Execution complete")
+            return target.background(_command, as_root=as_root)
 
-            logfile = ArtifactPath.join(self.res_dir, 'output.log')
-            logger.debug('Saving stdout to %s...', logfile)
+        self.output = target.execute(_command, as_root=as_root, timeout=timeout)
```

Comment:
Do we need `output` attribute outside of that function ? If possible, it would be better to only keep the "static configuration" in the state of the object and just let people look at the return of `run()` if they need it, rather than introducing extra mutable state that can be looked at when it's garbage (i.e. None).